tests/unit_tests/data/test_preprocess_data.py

```diff
@@ -67,7 +71,7 @@ def merge_datasets(idir):
     merge_main()
 
 
-def do_test_preprocess_data(temp_dir, extra_args=[]):
+def do_test_preprocess_data(temp_dir, extra_args=[], indexed_dataset_type=MMapIndexedDataset):
```

Comment:
No need for a default, let's make it a positional arg

Reply:
This argument is now a boolean called `s3` that defaults to False. Do you still want to make the new argument a positional one?